django/test/client.py

```diff
@@ -675,7 +684,17 @@ def _handle_redirects(self, response, **extra):
             if not path.startswith('/'):
                 path = urljoin(response.request['PATH_INFO'], path)
 
-            response = self.get(path, QueryDict(url.query), follow=False, **extra)
+            if response.status_code in (HTTPStatus.TEMPORARY_REDIRECT,
+                                        HTTPStatus.PERMANENT_REDIRECT):
+                # 307 and 308 status codes need to preserve their request method
```

Comment:
It is not the status code that needs to preserve the request method. Perhaps the following reads better:

`# Preserve request method post-redirect for 307 and 308 responses.`